src/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -63,5 +63,42 @@ public void Pop()
             Debug.Assert(_index > 0);
             Current = _previous[--_index];
         }
+
+        // Return a property path in the form of: [FullNameOfType].FirstProperty.SecondProperty.LastProperty
+        public string PropertyPath
+        {
+            get
+            {
+                StringBuilder path = new StringBuilder();
+
+                if (_previous == null || _index == 0)
+                {
+                    path.Append($"[{Current.JsonClassInfo.Type.FullName}]");
+                }
+                else
+                {
+                    path.Append($"[{_previous[0].JsonClassInfo.Type.FullName}]");
+
+                    for (int i = 0; i < _index; i++)
+                    {
+                        path.Append(GetPropertyName(_previous[i]));
+                    }
+                }
+
+                path.Append(GetPropertyName(Current));
+
+                return path.ToString();
+            }
+        }
+
+        private string GetPropertyName(in WriteStackFrame frame)
+        {
+            if (frame.JsonPropertyInfo != null && frame.JsonClassInfo.ClassType == ClassType.Object)
+            {
+                return $".{frame.JsonPropertyInfo.PropertyInfo.Name}";
+            }
+
+            return string.Empty;
```

Comment:
Do we want to avoid calling `Append("")` or is this only used in an exception path and hence perf isn't important?

Reply:
Currently yes `Path` is only used in exception cases. Also`Append("")` won't actually do any allocs or anything expensive if string is empty.